crates/sui-core/src/authority.rs

```diff
@@ -1529,14 +1531,23 @@ impl AuthorityState {
         .await
     }
 
+    pub async fn add_pending_certs_and_effects(
+        &self,
+        certs: Vec<VerifiedCertificate>,
+        effects: Vec<ValidTransactionEffects>,
+        digests: Vec<ValidExecutionDigests>,
+    ) -> SuiResult {
+        self.database
+            .store_pending_certs_and_effects(&certs, &effects, &digests)?;
+        let mut transaction_manager = self.transaction_manager.lock().await;
+        transaction_manager.enqueue(certs).await
```

Comment:
As discussed below, I think there are two approaches that are more consistent:
1. Pass both certs and effects as function args to `enqueue()`
2. Pass digest as the function arg to `enqueue()` and have `TransactionManager` load certs and effects from db. This is not as preferable as above.

What do you think?